core/src/main/scala/org/apache/spark/storage/BlockManager.scala

```diff
@@ -833,8 +833,10 @@ private[spark] class BlockManager(
     logDebug("Put block %s locally took %s".format(blockId, Utils.getUsedTimeMs(startTimeMs)))
 
     // Either we're storing bytes and we asynchronously started replication, or we're storing
-    // values and need to serialize and replicate them now:
-    if (putLevel.replication > 1) {
+    // values and need to serialize and replicate them now.
+    // Should not replicate the block if its StorageLevel is StorageLevel.NONE or
+    // putting it to local is failed.
+    if (!putBlockInfo.isFailed && putLevel.replication > 1) {
```

Comment:
i think @squito said is correct. now the problem is when block is failed in local and replication factor needs 2, but now how we choose to deal with this situation, there are three ways: 
one is making replication factor to 1 and continue to replication to remote node. 
two is stop to replication to remote node and its replication factor is 0. 
three is continuing to replication to two remote node and its replication factor is 2.
so i think the three maybe a good solution. how about your opinions? @andrewor14 @squito